### ISSUE TITLE
Use MediaRecorder?

### ISSUE THREAD

**Issue by jimmywarting** [binarykitchen/videomail-client#185]
It's seems like a lots of stuff going just to record a simple mediastream from the webcam by using the canvas element and working with frames

Guessing there is some frame drops cuz of all the heavy work that has to be manually done in javascript. Guess this was/is old technic when MediaRecorder wasn't around.

How about dropping all of that canvas stuff and use [MediaRecorder](https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder) instead?

**Comment by binarykitchen** [comment]
Yep, time for a big rewrite @jimmywarting - totally aware of it. Just lacking time.

Mind you, this videomail-client existed for more than 6 years when that Media Stream API wasn't available yet. Hence, all those hacks around canvas ;) 

Also, browser support for that API was poor over those years and evolved only very slowly ... with the existing solution I was ahead of everyone else during those years and were able to do to any cool stuff and customisations.

If I won $50k in lotto, I'd sit down and enjoy rewriting all that, grins.

**Comment by binarykitchen** [comment]
Still working on this ;)

At the same time, for mobile users, MediaRecorder, is very unstable, and we have made Videomail a lot more robust.
Keeping this ticket open for future ...

**Comment by binarykitchen** [comment]
@jimmywarting I'm planning to work on this soon in a new repository. Are you keen to join?
